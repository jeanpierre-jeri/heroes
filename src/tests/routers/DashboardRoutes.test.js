import DashboardRoutes from '../../routers/DashboardRoutes'
import { AuthContext } from '../../auth/authContext'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

describe('Pruebas en <DashBoardRoutes />', () => {
  const contextValue = {
    user: {
      logged: true,
      name: 'Juanito',
    },
  }

  test('debe de mostrarse correctamente', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.text-info').text().trim()).toBe('Juanito')
    expect(wrapper.find('h1').text().trim()).toBe('Marvel Screen')
  })

  test('debe de mostrarse correctamente de DC', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/dc']}>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h1').text().trim()).toBe('Dc Screen')
  })
})
