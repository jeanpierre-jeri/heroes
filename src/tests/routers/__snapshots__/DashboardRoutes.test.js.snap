// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashBoardRoutes /> debe de mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="navbar-item nav-link text-info"
        >
          Juanito
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 row-cols-lg-4 g-4 animate__animated animate__fadeIn"
      >
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Spider Man"
              className="card-img-top"
              src="/assets/marvel-spider.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Spider Man
              </h5>
              <p
                className="card-text"
              >
                Peter Parker
              </p>
              <p
                className="card-text"
              >
                Amazing Fantasy #15
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Captain America"
              className="card-img-top"
              src="/assets/marvel-captain.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Captain America
              </h5>
              <p
                className="card-text"
              >
                Steve Rogers
              </p>
              <p
                className="card-text"
              >
                Captain America Comics #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Iron Man"
              className="card-img-top"
              src="/assets/marvel-iron.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Iron Man
              </h5>
              <p
                className="card-text"
              >
                Tony Stark
              </p>
              <p
                className="card-text"
              >
                Tales of Suspense #39
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Thor"
              className="card-img-top"
              src="/assets/marvel-thor.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Thor
              </h5>
              <p
                className="card-text"
              >
                Thor Odinson
              </p>
              <p
                className="card-text"
              >
                Journey into Myster #83
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Hulk"
              className="card-img-top"
              src="/assets/marvel-hulk.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hulk
              </h5>
              <p
                className="card-text"
              >
                Bruce Banner
              </p>
              <p
                className="card-text"
              >
                The Incredible Hulk #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Wolverine"
              className="card-img-top"
              src="/assets/marvel-wolverine.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wolverine
              </h5>
              <p
                className="card-text"
              >
                James Howlett
              </p>
              <p
                className="card-text"
              >
                The Incredible Hulk #180
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Daredevil"
              className="card-img-top"
              src="/assets/marvel-daredevil.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Daredevil
              </h5>
              <p
                className="card-text"
              >
                Matthew Michael Murdock
              </p>
              <p
                className="card-text"
              >
                Daredevil #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Hawkeye"
              className="card-img-top"
              src="/assets/marvel-hawkeye.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hawkeye
              </h5>
              <p
                className="card-text"
              >
                Clinton Francis Barton
              </p>
              <p
                className="card-text"
              >
                Tales of Suspense #57
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Cyclops"
              className="card-img-top"
              src="/assets/marvel-cyclops.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Cyclops
              </h5>
              <p
                className="card-text"
              >
                Scott Summers
              </p>
              <p
                className="card-text"
              >
                X-Men #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Silver Surfer"
              className="card-img-top"
              src="/assets/marvel-silver.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Silver Surfer
              </h5>
              <p
                className="card-text"
              >
                Norrin Radd
              </p>
              <p
                className="card-text"
              >
                The Fantastic Four #48
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <DashBoardRoutes /> debe de mostrarse correctamente de DC 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="navbar-item nav-link text-info"
        >
          Juanito
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Dc Screen
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 row-cols-lg-4 g-4 animate__animated animate__fadeIn"
      >
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Batman"
              className="card-img-top"
              src="/assets/dc-batman.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                Detective Comics #27
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Superman"
              className="card-img-top"
              src="/assets/dc-superman.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Superman
              </h5>
              <p
                className="card-text"
              >
                Kal-El
              </p>
              <p
                className="card-text"
              >
                Action Comics #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-superman"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Flash"
              className="card-img-top"
              src="/assets/dc-flash.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Flash
              </h5>
              <p
                className="card-text"
              >
                Jay Garrick
              </p>
              <p
                className="card-text"
              >
                Flash Comics #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-flash"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
            <div
              className="card-footer"
            >
              <p
                className="text-muted"
              >
                Jay Garrick, Barry Allen, Wally West, Bart Allen
              </p>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Green Lantern"
              className="card-img-top"
              src="/assets/dc-green.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Green Lantern
              </h5>
              <p
                className="card-text"
              >
                Alan Scott
              </p>
              <p
                className="card-text"
              >
                All-American Comics #16
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-green"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
            <div
              className="card-footer"
            >
              <p
                className="text-muted"
              >
                Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
              </p>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Green Arrow"
              className="card-img-top"
              src="/assets/dc-arrow.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Green Arrow
              </h5>
              <p
                className="card-text"
              >
                Oliver Queen
              </p>
              <p
                className="card-text"
              >
                More Fun Comics #73
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-arrow"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Wonder Woman"
              className="card-img-top"
              src="/assets/dc-wonder.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wonder Woman
              </h5>
              <p
                className="card-text"
              >
                Princess Diana
              </p>
              <p
                className="card-text"
              >
                All Star Comics #8
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-wonder"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Martian Manhunter"
              className="card-img-top"
              src="/assets/dc-martian.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Martian Manhunter
              </h5>
              <p
                className="card-text"
              >
                J'onn J'onzz
              </p>
              <p
                className="card-text"
              >
                Detective Comics #225
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-martian"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
            <div
              className="card-footer"
            >
              <p
                className="text-muted"
              >
                Martian Manhunter
              </p>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Robin/Nightwing"
              className="card-img-top"
              src="/assets/dc-robin.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Robin/Nightwing
              </h5>
              <p
                className="card-text"
              >
                Dick Grayson
              </p>
              <p
                className="card-text"
              >
                Detective Comics #38
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-robin"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Blue Beetle"
              className="card-img-top"
              src="/assets/dc-blue.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Blue Beetle
              </h5>
              <p
                className="card-text"
              >
                Dan Garret
              </p>
              <p
                className="card-text"
              >
                Mystery Men Comics #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-blue"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
            <div
              className="card-footer"
            >
              <p
                className="text-muted"
              >
                Dan Garret, Ted Kord, Jaime Reyes
              </p>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Black Canary"
              className="card-img-top"
              src="/assets/dc-black.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Black Canary
              </h5>
              <p
                className="card-text"
              >
                Dinah Drake
              </p>
              <p
                className="card-text"
              >
                Flash Comics #86
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-black"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
            <div
              className="card-footer"
            >
              <p
                className="text-muted"
              >
                Dinah Drake, Dinah Lance
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
